### ISSUE TITLE
Request to add sklearn.impute.IterativeImputer

### ISSUE THREAD

**Issue by clgarciga** [pycaret/pycaret#471]
It's bound to come up eventually so I thought I'd get there first :-)

**Comment by pycaret** [comment]
@clgarciga Would you like to work on including KNN and Iterative imputer? The change will happen in `preprocess.py` with the documentation change in `setup` function for `classification.py` `regression.py` `clustering.py` `anomaly.py`.

Let me know if that works for you so I can assign it to you.

Thanks

**Comment by clgarciga** [comment]
@pycaret Still working on adding train/test split and CV options, and hoping to finish that up before working on something else. Come back to me later if no one else takes up this issue.

**Comment by pycaret** [comment]
@clgarciga ok

**Comment by zhenlingcn** [comment]
I want to add this feature into pycaret, but I find adding this function is too tricky because iterative imputer needs other values in the data. However, in the pipeline of pycaret, encoding occurs after imputation. Thus, we must deal with categorical data in the imputation process. 
A simple solution is just to use decision tree as the estimator and encoding all features by the ordinal encoder, but I do not ensure this solution is reasonable.

**Comment by Yard1** [comment]
@pycaret Do you think it would be possible to have encoding happen first?

**Comment by zhenlingcn** [comment]
I want to add another point. I think that fully supporting iterative Imputer is very complex because we not only need to encoding data but also need to normalize the data if we do not use the tree-based imputer. Thus, I think that implementing a specific instance (tree-based imputer) may be a more feasible way.

**Comment by pycaret** [comment]
@clgarciga @zhenlingcn @aliagowani 

This issue has been fixed in the `pycaret-nightly` release. Please install pycaret-nightly using `pip install pycaret-nightly`. Feel free to open a new issue, in case the problem persists. This fix will be published to PyPi as `pycaret==2.2.0` on October 28, 2020.
